dipy/nn/cnn_1d_denoising/cnn_1d_denoising.py

```diff
@@ -0,0 +1,312 @@
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras.layers import Conv1D, Activation
+from tensorflow.keras.models import Model
+from tensorflow.keras.layers import Input
+from sklearn.model_selection import train_test_split
+'''
+Title : Denoising diffusion weighted imaging data using CNN
+=====
+Obtaining tissue microstructure measurements from diffusion weighted imaging (DWI) with multiple,
+high b-values is crucial. However, the high noise levels present in these images can adversely affect
+the accuracy of the microstructural measurements. In this context, we suggest a straightforward denoising technique that
+can be applied to any DWI dataset as long as a low-noise, single-subject dataset is obtained using the same DWI sequence.
+We created a simple 1D-CNN model with five layers, based on the 1D CNN for denoising speeches.
+The model consists of two convolutional layers followed by max-pooling layers, and a dense layer.
+The first convolutional layer has 16 one-dimensional filters of size 16, and the second layer has 32 filters of size 8.
+ReLu activation function is applied to both convolutional layers. The max-pooling layer has a kernel
+size of 2 and a stride of 2. 
+The dense layer maps the features extracted from the noisy image to the low-noise reference image.
+
+Reference:
+---------
+Denoising diffusion weighted imaging data using convolutional neural networks
+Hu Cheng, Sophia Vinci-Booher, Jian Wang, Bradley Caron, Qiuting Wen, Sharlene Newman, Franco Pestilli
```

Comment:
Please refer to this PLoS one paper: https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0274396

Reply:
Added PLoS